packages/snfoundry/contracts/src/YourCollectible.cairo

```diff
@@ -74,8 +76,8 @@ mod YourCollectible {
         owned_tokens_index: LegacyMap<u256, u256>,
         // Mapping with all token ids, 
         all_tokens: LegacyMap<u256, u256>,
-        // Counter for all tokens, used for enumeration
-        all_tokens_counter: u256,
+        // Helper to get the length of `all_tokens`
+        all_tokens_length: u256,
```

Comment:
whats the difference between `all_tokens_length` and `token_id_counter` , i think this implementation needs more simplification. 

Also why do we need to override `total_supply`

Reply:
1. Most of the cases `all_tokens_length` and `token_id_counter` will have the same value, but in case of `burn` token, `all_tokens_length` will decrease by 1 and `token_id_counter` wont be affected. See https://github.com/Quantum3-Labs/speedrunstark/blob/958b0c883305acdff358ac354ad152389350b49e/packages/snfoundry/contracts/src/YourCollectible.cairo#L203

2. We are not overriding `total_supply`, actually oz doesnt provide `total_supply` for erc721. So we have to implement it.

3. In the next PR I will try to simply more, specially to remove the `ERC721ReceiverComponent`